armi/utils/plotting.py

```diff
@@ -1423,3 +1417,147 @@ def plotBlockDiagram(block, fName, cold, cmapName="RdYlBu", materialList=None):
+    hexSideSubdivisions : int, optional
+        By default, it is 1 so that the triangle edge length is 1 if hexPitch=sqrt(3).
+    imgFileExt : str, optional
+        The image file extension.
+
+    Examples
+    --------
+    >>> rtflux = rtflux.RtfluxStream.readBinary("RTFLUX")
+    >>> plotTriangleFlux(rtflux, axialZ=10, energyGroup=4)
+    """
+    triHeightInCm = hexPitch / 2.0 / hexSideSubdivisions
+    sideLengthInCm = triHeightInCm / (math.sqrt(3.0) / 2.0)
+    s2InCm = sideLengthInCm / 2.0
+
+    vals = rtfluxData.groupFluxes[:, :, axialZ, energyGroup]
+    patches = []
+    colorVals = []
+    for i in range(vals.shape[0]):
+        for j in range(vals.shape[1]):
+            # use (i+j)%2 for rectangular meshing
+            flipped = i % 2
+            xInCm = s2InCm * (i - j)
+            yInCm = triHeightInCm * j + sideLengthInCm / 2.0 / math.sqrt(3) * (
+                1 + flipped
+            )
+
+            flux = vals[i][j]
+
+            if flux:
+                triangle = patches.mpatches.RegularPolygon(
+                    (xInCm, yInCm),
+                    3,
+                    sideLengthInCm / math.sqrt(3),
+                    orientation=math.pi * flipped,
+                    linewidth=0.0,
+                )
+
+                patches.append(triangle)
+                colorVals.append(flux)
+
+    collection = PatchCollection(patches, alpha=1.0, linewidths=(0,), edgecolors="none")
+    # add color map to this collection ONLY (pins, not ducts)
+    collection.set_array(numpy.array(colorVals))
+
+    plt.figure()
+    ax = plt.gca()
+    ax.add_collection(collection)
+    colbar = plt.colorbar(collection)
+    colbar.set_label("n/s/cm$^3$")
+    plt.ylabel("cm")
+    plt.xlabel("cm")
+    ax.autoscale_view()
+    plt.savefig("RTFLUX-z" + str(axialZ + 1) + "-g" + str(energyGroup + 1) + imgFileExt)
+    plt.close()
+
+
+def plotNucXs(
+    isotxs, nucNames, xsNames, fName=None, label=None, noShow=False, title=None
+):
+    """
+    generates a XS plot for a nuclide on the ISOTXS library
+
+    Parameters
+    ----------
+    isotxs : IsotxsLibrary
+        A collection of cross sections (XS) for both neutron and gamma reactions.
+    nucName : str or list
+        The nuclides to plot
+    xsName : str or list
+        the XS to plot e.g. n,g, n,f, nalph, etc. see xsCollections for actual names.
+    fName : str, optional
+        if fName is given, the file will be written rather than plotting to screen
+    label : str, optional
+        is an optional label for image legends, useful in ipython sessions.
+    noShow : bool, optional
+        Won't finalize plot. Useful for using this to make custom plots.
+
+    Examples
+    --------
+    >>> l = ISOTXS()
+    >>> l.plotNucXs('U238NA','fission')
+
+    >>> # Plot n,g for all xenon and krypton isotopes
+    >>> f = lambda name: 'XE' in name or 'KR' in name
+    >>> l.plotNucXs(sorted(filter(f,l.nuclides.keys())),itertools.repeat('nGamma'))
```

Comment:
I think you need to update these examples to reflect that this is no longer a method, but is not a function. i.e. you need to pass in ISOTXS explicitly.

Reply:
Ah! You are totally right. I will look and see if I did this elsewhere to.  Good catch.